### PR TITLE
Updates file generation in create command

### PR DESCRIPTION
**Pull request by kinetifex**
Moving away from `vinyl-fs` due to outdated dependencies with CVEs and no end in sight for updates.

The updated solution also keeps track of overridden files and notes the file plugin source in output.

Screen Shot 2022-05-20 at 8 56 40 AM

## Changelog

**@gasket/cli**
- Drop `vinyl-fs` and show final plugin source for files that were overridden

## Test Plan

- Updated unit tests
- locally tested (see above screenshot)

### CODE REVIEW

File: packages/gasket-cli/src/scaffold/actions/generate-files.js [hunk 1]
```diff
@@ -1,78 +1,128 @@
...
-async function generateFiles(context, spinner) {
-  const { dest, files, generatedFiles } = context;
+async function getDescriptors(context) {
+  const { dest, files } = context;
+  const descriptors = (
+    await Promise.all(files.globSets.map(async set => {
+      const { globs, source } = set;
+      const matches = await Promise.all(globs.map(async pattern => {
+        const srcPaths = await glob(pattern, { nodir: true });
+        const base = path.resolve(pattern.replace(/\/(.)?\*(.+)?/, ''));
```
Comment (jpage-godaddy) [inline packages/gasket-cli/src/scaffold/actions/generate-files.js:50]: What is this pattern supposed to match? Is that `(.)` correct or is it supposed to be:

```suggestion
        const base = path.resolve(pattern.replace(/\/\.?\*.*/, ''));
```

File: packages/gasket-cli/package.json [hunk 2]
```diff
@@ -76,14 +76,12 @@
...
     "json5": "^2.1.0",
     "lodash.defaultsdeep": "^4.6.1",
-    "map-stream": "^0.1.0",
     "mkdirp": "^1.0.4",
     "ora": "^3.4.0",
     "pump": "^3.0.0",
     "semver": "^7.3.4",
     "tar-fs": "^2.0.0",
-    "untildify": "^4.0.0",
-    "vinyl-fs": "^3.0.3"
```
Comment (cwang-godaddy) [inline packages/gasket-cli/package.json:86]: 👍 this also drops one of the vulnerability `glob-parent@3.1.0`
```
│ └─┬ vinyl-fs@3.0.3
│   └─┬ glob-stream@6.1.0
│     └── glob-parent@3.1.0
```

File: packages/gasket-cli/src/scaffold/actions/generate-files.js [hunk 3]
```diff
@@ -1,78 +1,128 @@
...
+            srcFile: srcPaths[idx],
+            targetFile: path.join(dest, target),
+            target,
+            from: source.name
+          };
+        });
+      }));
+      return matches.flat();
+    }))
+  ).flat();
```
[the remark this answers is not in the document]
Comment (kinetifex) [reply packages/gasket-cli/src/scaffold/actions/generate-files.js:65]: I guess not officially - I'll adjust these. Good catch

**Review by jpage-godaddy — APPROVED** [review]

**Review by rxmarbles — APPROVED** [review]

File: packages/gasket-cli/package.json (hunk 2 above)
Comment (agerard-godaddy) [reply packages/gasket-cli/package.json:86]: 👍 Exactly - the main reason instigating this change.

**Review by kawikabader — APPROVED** [review]

**Pull request merged**